src/AwsClient.php

```diff
@@ -275,6 +276,7 @@ public function __construct(array $args)
 
         if (!is_null($this->api->getMetadata('awsQueryCompatible'))) {
```

Comment:
I think is safer if this condition is scoped inside of the addQueryModeHeader method.
For example:
```php
private function addQueryModeHeader(): void
{
    if (is_null($this->api->getMetadata('awsQueryCompatible'))) {
            return;
    }
    
    ...add the header.
}
```

Reply:
I've thought about it and for readability and to avoid looking for this trait more than once for each client instantiation, I'd like to keep the changes as they are now